### DISCUSSION TITLE
Metapost via gmp

### DISCUSSION THREAD

**Discussion by openscript** [tectonic-typesetting/tectonic#895] [Q&A] [answered]
Hello

Thank you for the great effort!

Is it possible to include [gmp](https://ctan.org/pkg/gmp) in https://data1.fullyjustified.net/tlextras-2021.3r1.tar?

This would be great as it enables the usage of Metapost.

Robin

**Comment by pkgw** [comment, marked as answer]
It is definitely possible, but I am not sure how useful it would be. Tectonic creates PDF files directly from the XeTeX engine's XDV output using its variant of `xdvipdmx`, which removes all of the code that executes GhostScript ­— because a major goal of Tectonic is to promote reliable, reproducible document builds by avoiding dependencies on external tools. So I am pretty sure that there isn't a way for Tectonic to incorporate MetaPost output in the documents it creates, since (as I understand it) the latter is all about PostScript, and Tectonic/xdvipdfmx need GhostScript to do anything with PostScript content. Tectonic is more limited than XeLaTeX in this regard due to the separation from GhostScript.

All that being said, this isn't actually something I've tried playing around with, so perhaps there are aspects that would be useful even given the above limitations?

**Reply by openscript** [reply]
Thank you for elaborating on this. I've learned something.

What can be an easy work around for this? Maybe using `\write18` for generating a pdf and include it later?

**Reply by pkgw** [reply]
If I'm understanding your question correctly ... My general feeling is that it's much better to have a step in your workflow where you create your figure files in the right format(s) that you need, then feed them into Tectonic as the final stage. So you'd want to figure out some workflow for turning your MetaPost content into PDF figures. To stick to a workflow that used *only* Tectonic for TeX processing, I think you'd need to compile MetaPost figured to XDV, use real `xdvipdfmx`, then assemble into a final PDF with Tectonic. Or, compile the MetaPost content with a stock TeXLive engine.
